examples/simulate_fight_scenario.py

```diff
@@ -0,0 +1,111 @@
+from sc2 import maps
+from sc2.bot_ai import BotAI
+from sc2.data import Race, Difficulty
+from sc2.ids.unit_typeid import UnitTypeId
+from sc2.main import run_game
+from sc2.player import Bot, Computer
+from loguru import logger
+
+
+class FightBot(BotAI):
+    def __init__(self):
+        super().__init__()
+        self.control_requested = False
+        self.control_received = False
+        self.fight_started = False
+        self.supplies_been_damaged = False
+
+    async def on_step(self, iteration):
+        # prepare the level
+        if not self.control_requested:
+            # we need this one for `self.enemy_units` to "see" all units
+            await self._client.debug_show_map()
+            # this one will allow us to do something like: `self.enemy_units.first.attack(self.townhalls.first)`
+            await self._client.debug_control_enemy()
+            logger.info("control requested")
+            # await self.chat_send("control requested")
+            self.control_requested = True
+
+        if self.control_requested and self.enemy_units and not self.control_received:
+            # prepare my side
+            me = 1
+            cc = self.townhalls.first
```

Comment:
Since this is played on flat64, and there are no normal spawns, there will probably be no townhalls and so this will crash.